pkg/kubelet/kubelet.go

```diff
@@ -432,6 +433,10 @@ func (kl *Kubelet) Run(updates <-chan PodUpdate) {
 	if kl.dockerPuller == nil {
 		kl.dockerPuller = dockertools.NewDockerPuller(kl.dockerClient, kl.pullQPS, kl.pullBurst)
 	}
+	if kl.dockerCache == nil {
+		kl.dockerCache = dockertools.NewDockerCache(kl.dockerClient)
+	}
+	go util.Forever(func() { kl.dockerCache.UpdateContainers() }, 50*time.Millisecond)
```

Comment:
This incurs constant overhead on the Kubelet and the Docker daemon, but does make the latency of the request 0. We should reduce it during period when the Kubelet is just sitting idle.

WDYT of the naive cache with a timeout? Maybe we can even schedule refreshes during that timeout. It seems like the first request may incur higher latency, but the remaining ones would not. This seems to match the case we're trying to improve.